File: API/Contracts/GiveOrderRequest.cs
using System;
using System.ComponentModel;
using Core.Data.Entity;

namespace Core.Features.GiveOrder
{
    public class GiveOrderRequest
    {
        [DefaultValue("3aa42229-1c0f-4630-8c1a-db879ecd0427")]
        public Guid? UserId { get; set; }

        [DefaultValue("AAPL")]
        public string? StockSymbol { get; set; }

        [DefaultValue(1)]
        public int? Quantity { get; set; }

        [DefaultValue(150.0)]
        public decimal? TargetPrice { get; set; }

        [DefaultValue(OrderType.Buy)]
        public OrderType? OrderType { get; set; }
    }
}

File: frontend/src/components/GiveOrderRequest.js
import axios from 'axios';

const giveOrderRequest = async (userId, stockSymbol, quantity, targetPrice, orderType) => {
  const orderTypeInt = orderType === 'Buy' ? 0 : 1;
  try {
    const response = await axios.post('http://localhost:5244/api/orders/place', {
      userId: userId,
      stockSymbol: stockSymbol,
      quantity: quantity,
      targetPrice: targetPrice,
      orderType: orderTypeInt,
    });
    return response.data;
  } catch (error) {
    console.error("Error placing order:", error);
    throw error;
  }
};

export { giveOrderRequest };
